lib/java/com/google/android/material/snackbar/BaseTransientBottomBar.java

```diff
@@ -364,6 +366,21 @@ private Drawable createThemedBackground() {
     return background;
   }
 
+  void updateThemedBackgroundOverlayColor(@ColorInt int overlayColor) {
+    int backgroundColor = MaterialColors.getColor(view, R.attr.colorSurface);
+    updateThemedBackgroundColors(backgroundColor, overlayColor);
+  }
+
+  private void updateThemedBackgroundColors(@ColorInt int backgroundColor, @ColorInt int overlayColor) {
+    if (!(view.getBackground() instanceof GradientDrawable)) return;
+    int color =
```

Comment:
@ricknout Will they layering of `MaterialColors` work properly with this solution? I mean if for example one uses a blue color as a `colorSurface` and applies a red color as `backgroundColor` would it still be a desirable result?

Reply:
Hmm not so sure. I copied what was done in `BaseTransientBottomBar#createThemedBackground` for consistency, so I assume it would happen either way

Reply:
@ricknout Okay thanks for explaining! I was just wondering if the alternative provided on the issue you linked, would be better, since this is a direct user action and the color they would like to apply is the one they passed to that method. That's just a thought I was having, but I a still learning the Material internals, so there might be a specific reasoning behind that color layering that I am not aware of :) Thanks again for jumping on the issue and providing a solution so quickly!

Reply:
@pavlospt No problem :-) The alternative in the issue might indeed be better, interested in what the MDC team thinks

Reply:
Thanks for the PR! I think the alternative (a getBackground() or setBackgroundTint() method) is preferable because the layering of `colorSurface` and `colorOnSurface` is just meant to provide the default behavior according to the Material spec. If the client wants to override that behavior I think they would probably want to pass in a color that will be used for the background without any layering, and if they do want some layering they could do it before passing in the color.

Reply:
@dsn5ft okay great, thanks for the feedback. I'll close this PR and open a new one with these changes, so that the commit history is clean

Reply:
Np! No need to close the PR if you don't want to. We'll submit this change internally anyway and then this PR will get automatically closed when the commit is synced to GitHub. So the commit history from this PR won't make it to master.

Reply:
Ah okay! Will just submit an update then

Reply:
Okay @dsn5ft I've updated the PR. I had to add in a few workarounds to ensure the tint works on all API levels (see the comment in the method)

Reply:
looks good!